### DISCUSSION TITLE
Выключение по времени.

### DISCUSSION THREAD

**Discussion by vAkUuMkA** [vvip-68/LedPanelWiFi#79] [Вопросы и ответы]
Здравствуйте. В очередной раз собрал этот проект. Спасибо огромное за Ваш труд. Всё отлично работает, только никак не получается заставить гирлянду выключаться по времени. В приложении в «режимах» ставлю время включения и выключения. По времени включается, но не выключается. Ставлю к примеру в 17:00 включить и в 23:00 выключить и в 05:00 включить и 07:30 выключить. Или надо в другом порядке выставлять время? Спасибо.

**Comment by vvip-68** [comment]
Да нет - должно быть без разницы в каком порядке
Имея такой разброс времени естественно на проверку уйдут сутки.
Я проверил последовательно задавая вкл/выкл через две минуты - все работает.

- Попробуйте отсортировать время по возрастанию, указывая соответственно времени что - включать или выключать
- Попробуйте вместо "Выключить" поставить "Ночные часы"...

В общем - я никаких проблем с работоспособностью режимов не обнаружил.

**Reply by vAkUuMkA** [reply]
Спасибо за ответ. Вроде разобрался - конечно во всё виноват сам. Всё корректно включает и отключает по времени. Просто она потом включалась так как сам включил включение "Демо режима" через минуту.  А вот ещё вопрос по поводу вывода погоды на другом языке. Я не создавал новый язык, а просто перевёл даты, время и погодные условия  в a_def_lang. И на всякий случай во всех трёх языках что там есть (русский, английский и испанский) Погоду беру в Openweather. 
Запрос текущей погоды
[HTTP] GET...
[HTTP] GET... code: 200
Погода получена: OpenWeatherMap
Город: Kandava
Сейчас: scattered clouds, +6ºC
Код иконки: '03d'
Код погоды: 802
Но в бегущей строке пишет погоду на английском. Где я опять накосячил?

**Reply by vvip-68** [reply]
Ничё не понял. Какой язык выбран основным? А какой нужен?
Там дело в чем - в зависимости от выбранного языка, в запросе погоды указывается локаль, на которой OpenWeatherMap и присылает ответы. Это в Weather.ino примерно здесь
![image](https://github.com/user-attachments/assets/ae8747b4-0fdf-4ac5-bfff-7b58cf0fe491)

Это значит, что тут либо надо переопределить константу  WTR_LANG_OWM на нужную локаль (у вас же 'lv'?), либо явно ее прописать в запросе, либо вот тут закомментарить строчку

![image](https://github.com/user-attachments/assets/dc48aa26-a709-40d0-bd75-bd32d469a93d)

**Reply by vvip-68** [reply]
Если вы выполнили перевод на другой язык - может поделитесь переводом? Я включу его в комплект поставки проекта...

**Reply by vvip-68** [reply]
Похоже, OWM не понимает латышский - все равно возвращает английский текст. Значит выход - закомментировать вот ту строчку, что на второй картинке

**Reply by vAkUuMkA** [reply]
Без проблем, я тогда отредактирую перевод, проверю у умных людей и перешлю Вам. 
По поводу закомментировать это Вы про 455 строчку? Уж простите за ламмерские вопросы, но хочется немного разобраться. Я думал что если я всё перевёл в языке по умолчанию на латышский и потом получаю к примеру ответ от OWM: Код погоды: 802 то на экран будет выведена та переведённая 195 строка
  static const char W_CODE_802[] PROGMEM = "Mainīgs mākoņainums";                // scattered clouds: 25-50%

**Reply by vvip-68** [reply]
Строчка 455 (которую нужно закомментировать) означает - если от сервера получен ответ в виде строки (в вашем логе ответ получен - "scattered clouds" - ничего перекодировать не нужно - выводить "как есть". А есть - на английском.
 Закомментируете - будет пользоваться переводом из файла a_def_lang.h

**Reply by vAkUuMkA** [reply]
Спасибо, будем пробовать. Как закончу с переводом то перешлю его Вам.

**Reply by vvip-68** [reply]
И на какие грабли можете еще напороться - вот этих "специфических букв" в строке - вполне может и не быть в шрифтах (я не помню и не проверялось). Тогда на месте этих букв будет дырка. Или надо переписывать код знакогенератора для латышского языка...  Хотя, вроде, я уже с кем-то общался из Латвии и именно для этого языка добавлялись символы. Я надеюсь , что это так...

**Reply by vAkUuMkA** [reply]
Вот и посмотрим. И тогда придётся углубляться глубже. Часть букв даёт. Но есть и пустые места. Я подумал что это из за строчных букв. Оказывается дело серъёзнее)

**Reply by vAkUuMkA** [reply]
Вот вроде закончил с переводом. Конечно не претендую на лавры, но в любом случае лучше чем ничего. А как передать? Ну а со шрифтами буду попозже разбираться. Сейчас в бегущей строке показываются буквы литовского языка. Но он отличается от латышского.

**Reply by vvip-68** [reply]
Перевод передать - ну либо сюда прикрепить файл, либо стукнуться в личку https://vk.com/iivanovii
И желательно именно с написанием латышскими буквами, как правильно.
По буквам. Напиши здесь каких не хватает. Вот прям все перечисли здесь и заглавные и строчные.
